@iocanel the CI failed to compile
```
Caused by: org.apache.maven.plugin.compiler.CompilationFailureException: Compilation failure
/home/runner/work/quarkus/quarkus/extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageProcessor.java:[39,13] cannot find symbol
  symbol:   variable UNKNONW_USER
  location: class io.quarkus.container.image.deployment.ContainerImageProcessor
```

--- a/extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageProcessor.java
+++ b/extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageProcessor.java
@@ -3,17 +3,24 @@ package io.quarkus.container.image.deployment;
 import java.util.Collections;
 import java.util.Optional;
 
+import org.jboss.logging.Logger;
+
 import io.quarkus.container.spi.ContainerImageInfoBuildItem;
 import io.quarkus.container.spi.ImageReference;
 import io.quarkus.deployment.Capabilities;
+import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.ApplicationInfoBuildItem;
 
 public class ContainerImageProcessor {
 
+    private static final String UNKNOWN_USER = "?";
+    private static final Logger log = Logger.getLogger(ContainerImageProcessor.class);
+
     @BuildStep
-    public ContainerImageInfoBuildItem publishImageInfo(ApplicationInfoBuildItem app,
-            ContainerImageConfig containerImageConfig, Capabilities capabilities) {
+    public void publishImageInfo(ApplicationInfoBuildItem app,
+            ContainerImageConfig containerImageConfig, Capabilities capabilities,
+            BuildProducer<ContainerImageInfoBuildItem> containerImage) {
 
         ensureSingleContainerImageExtension(capabilities);
 
@@ -27,12 +34,21 @@ public class ContainerImageProcessor {
             }
         }
 
+        String effectiveGroup = containerImageConfig.getEffectiveGroup().orElse("");
+        //This can be the case when running inside kubernetes/minikube in dev-mode. Instead of failing, we should just catch and log.
+        if (UNKNOWN_USER.equals(effectiveGroup)) {
+            log.warn(
+                    "Can't get the current user name, which is used as default for container image group. Can't publish container image info.");
+            return;
+        }
+
         // if the user supplied the entire image string, use it
         if (containerImageConfig.image.isPresent()) {
             ImageReference imageReference = ImageReference.parse(containerImageConfig.image.get());
             String repository = imageReference.getRepository();
-            return new ContainerImageInfoBuildItem(Optional.of(imageReference.getRegistry()), repository,
-                    imageReference.getTag(), containerImageConfig.additionalTags.orElse(Collections.emptyList()));
+            containerImage.produce(new ContainerImageInfoBuildItem(Optional.of(imageReference.getRegistry()), repository,
+                    imageReference.getTag(), containerImageConfig.additionalTags.orElse(Collections.emptyList())));
+            return;
         }
 
         String registry = containerImageConfig.registry.orElse(null);
@@ -44,7 +60,7 @@ public class ContainerImageProcessor {
         String repository = (containerImageConfig.getEffectiveGroup().map(s -> s + "/").orElse("")) + effectiveName;
         if (!ImageReference.isValidRepository(repository)) {
             throw new IllegalArgumentException("The supplied combination of container-image group '"
-                    + containerImageConfig.getEffectiveGroup().orElse("") + "' and name '" + effectiveName + "' is invalid");
+                    + effectiveGroup + "' and name '" + effectiveName + "' is invalid");
         }
 
         final String effectiveTag = containerImageConfig.tag.orElse(app.getVersion());
@@ -52,10 +68,10 @@ public class ContainerImageProcessor {
             throw new IllegalArgumentException("The supplied container-image tag '" + effectiveTag + "' is invalid");
         }
 
-        return new ContainerImageInfoBuildItem(containerImageConfig.registry,
+        containerImage.produce(new ContainerImageInfoBuildItem(containerImageConfig.registry,
                 containerImageConfig.getEffectiveGroup(),
                 effectiveName, effectiveTag,
-                containerImageConfig.additionalTags.orElse(Collections.emptyList()));
+                containerImageConfig.additionalTags.orElse(Collections.emptyList())));
     }
 
     private void ensureSingleContainerImageExtension(Capabilities capabilities) {

--- a/extensions/kubernetes/minikube/deployment/src/main/java/io/quarkus/minikube/deployment/MinikubeProcessor.java
+++ b/extensions/kubernetes/minikube/deployment/src/main/java/io/quarkus/minikube/deployment/MinikubeProcessor.java
@@ -111,7 +111,7 @@ public class MinikubeProcessor {
         List<DecoratorBuildItem> result = new ArrayList<>();
         String name = ResourceNameUtil.getResourceName(config, applicationInfo);
 
-        Project project = KubernetesCommonHelper.createProject(applicationInfo, outputTarget, packageConfig);
+        Optional<Project> project = KubernetesCommonHelper.createProject(applicationInfo, outputTarget, packageConfig);
         result.addAll(KubernetesCommonHelper.createDecorators(project, MINIKUBE, name, config, metricsConfiguration,
                 annotations, labels, command,
                 ports, livenessPath, readinessPath, roles, roleBindings));

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KnativeProcessor.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KnativeProcessor.java
@@ -106,7 +106,7 @@ public class KnativeProcessor {
         List<DecoratorBuildItem> result = new ArrayList<>();
         String name = ResourceNameUtil.getResourceName(config, applicationInfo);
 
-        Project project = KubernetesCommonHelper.createProject(applicationInfo, outputTarget, packageConfig);
+        Optional<Project> project = KubernetesCommonHelper.createProject(applicationInfo, outputTarget, packageConfig);
         result.addAll(KubernetesCommonHelper.createDecorators(project, KNATIVE, name, config, metricsConfiguration, annotations,
                 labels, command,
                 ports, livenessPath, readinessPath, roles, roleBindings));

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java
@@ -70,23 +70,28 @@ public class KubernetesCommonHelper {
 
     private static final String OUTPUT_ARTIFACT_FORMAT = "%s%s.jar";
 
-    public static Project createProject(ApplicationInfoBuildItem app, OutputTargetBuildItem outputTarget,
+    public static Optional<Project> createProject(ApplicationInfoBuildItem app, OutputTargetBuildItem outputTarget,
             PackageConfig packageConfig) {
         return createProject(app, outputTarget.getOutputDirectory()
                 .resolve(String.format(OUTPUT_ARTIFACT_FORMAT, outputTarget.getBaseName(), packageConfig.runnerSuffix)));
     }
 
-    public static Project createProject(ApplicationInfoBuildItem app, Path artifactPath) {
+    public static Optional<Project> createProject(ApplicationInfoBuildItem app, Path artifactPath) {
         //Let dekorate create a Project instance and then override with what is found in ApplicationInfoBuildItem.
-        Project project = FileProjectFactory.create(artifactPath.toFile());
-        BuildInfo buildInfo = new BuildInfo(app.getName(), app.getVersion(),
-                "jar", project.getBuildInfo().getBuildTool(),
-                project.getBuildInfo().getBuildToolVersion(),
-                artifactPath.toAbsolutePath(),
-                project.getBuildInfo().getClassOutputDir(),
-                project.getBuildInfo().getResourceDir());
+        try {
+            Project project = FileProjectFactory.create(artifactPath.toFile());
+            BuildInfo buildInfo = new BuildInfo(app.getName(), app.getVersion(),
+                    "jar", project.getBuildInfo().getBuildTool(),
+                    project.getBuildInfo().getBuildToolVersion(),
+                    artifactPath.toAbsolutePath(),
+                    project.getBuildInfo().getClassOutputDir(),
+                    project.getBuildInfo().getResourceDir());
 
-        return new Project(project.getRoot(), buildInfo, project.getScmInfo());
+            return Optional.of(new Project(project.getRoot(), buildInfo, project.getScmInfo()));
+
+        } catch (Exception e) {
+            return Optional.empty();
+        }
     }
 
     /**
@@ -112,7 +117,7 @@ public class KubernetesCommonHelper {
     /**
      * Creates the common decorator build items.
      */
-    public static List<DecoratorBuildItem> createDecorators(Project project, String target, String name,
+    public static List<DecoratorBuildItem> createDecorators(Optional<Project> project, String target, String name,
             PlatformConfiguration config,
             Optional<MetricsCapabilityBuildItem> metricsConfiguration,
             List<KubernetesAnnotationBuildItem> annotations,
@@ -177,7 +182,7 @@ public class KubernetesCommonHelper {
      * @param name The name of the resource to accept the configuration
      * @param config The {@link PlatformConfiguration} instance
      */
-    private static List<DecoratorBuildItem> createContainerDecorators(Project project, String target, String name,
+    private static List<DecoratorBuildItem> createContainerDecorators(Optional<Project> project, String target, String name,
             PlatformConfiguration config) {
         List<DecoratorBuildItem> result = new ArrayList<>();
         if (config.getNamespace().isPresent()) {
@@ -206,7 +211,7 @@ public class KubernetesCommonHelper {
      * @param name The name of the resource to accept the configuration
      * @param config The {@link PlatformConfiguration} instance
      */
-    private static List<DecoratorBuildItem> createPodDecorators(Project project, String target, String name,
+    private static List<DecoratorBuildItem> createPodDecorators(Optional<Project> project, String target, String name,
             PlatformConfiguration config) {
         List<DecoratorBuildItem> result = new ArrayList<>();
         config.getImagePullSecrets().ifPresent(l -> {
@@ -248,7 +253,8 @@ public class KubernetesCommonHelper {
         return result;
     }
 
-    private static List<DecoratorBuildItem> createMountAndVolumeDecorators(Project project, String target, String name,
+    private static List<DecoratorBuildItem> createMountAndVolumeDecorators(Optional<Project> project, String target,
+            String name,
             PlatformConfiguration config) {
         List<DecoratorBuildItem> result = new ArrayList<>();
         config.getMounts().entrySet().forEach(e -> {
@@ -282,30 +288,32 @@ public class KubernetesCommonHelper {
         return result;
     }
 
-    private static List<DecoratorBuildItem> createAnnotationDecorators(Project project, String target, String name,
+    private static List<DecoratorBuildItem> createAnnotationDecorators(Optional<Project> project, String target, String name,
             PlatformConfiguration config,
             Optional<MetricsCapabilityBuildItem> metricsConfiguration,
             List<KubernetesPortBuildItem> ports) {
         List<DecoratorBuildItem> result = new ArrayList<>();
         ZonedDateTime now = ZonedDateTime.now(ZoneOffset.UTC);
 
-        ScmInfo scm = project.getScmInfo();
-        String vcsUrl = scm != null ? scm.getUrl() : null;
-        String commitId = scm != null ? scm.getCommit() : null;
+        project.ifPresent(p -> {
+            ScmInfo scm = p.getScmInfo();
+            String vcsUrl = scm != null ? scm.getUrl() : null;
+            String commitId = scm != null ? scm.getCommit() : null;
 
-        //Dekorate uses its own annotations. Let's replace them with the quarkus ones.
-        result.add(new DecoratorBuildItem(target, new RemoveAnnotationDecorator(Annotations.VCS_URL)));
-        result.add(new DecoratorBuildItem(target, new RemoveAnnotationDecorator(Annotations.COMMIT_ID)));
+            //Dekorate uses its own annotations. Let's replace them with the quarkus ones.
+            result.add(new DecoratorBuildItem(target, new RemoveAnnotationDecorator(Annotations.VCS_URL)));
+            result.add(new DecoratorBuildItem(target, new RemoveAnnotationDecorator(Annotations.COMMIT_ID)));
 
-        //Add quarkus vcs annotations
-        if (commitId != null) {
-            result.add(new DecoratorBuildItem(target,
-                    new AddAnnotationDecorator(name, new Annotation(QUARKUS_ANNOTATIONS_COMMIT_ID, commitId, new String[0]))));
-        }
-        if (vcsUrl != null) {
-            result.add(new DecoratorBuildItem(target,
-                    new AddAnnotationDecorator(name, new Annotation(QUARKUS_ANNOTATIONS_VCS_URL, vcsUrl, new String[0]))));
-        }
+            //Add quarkus vcs annotations
+            if (commitId != null) {
+                result.add(new DecoratorBuildItem(target, new AddAnnotationDecorator(name,
+                        new Annotation(QUARKUS_ANNOTATIONS_COMMIT_ID, commitId, new String[0]))));
+            }
+            if (vcsUrl != null) {
+                result.add(new DecoratorBuildItem(target,
+                        new AddAnnotationDecorator(name, new Annotation(QUARKUS_ANNOTATIONS_VCS_URL, vcsUrl, new String[0]))));
+            }
+        });
 
         if (config.isAddBuildTimestamp()) {
             result.add(new DecoratorBuildItem(target,

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java
@@ -13,6 +13,7 @@ import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.Set;
 import java.util.stream.Collectors;
 
@@ -116,87 +117,95 @@ class KubernetesProcessor {
                 .resolve(String.format(OUTPUT_ARTIFACT_FORMAT, outputTarget.getBaseName(), packageConfig.runnerSuffix));
 
         try {
-            final Map<String, String> generatedResourcesMap;
             // by passing false to SimpleFileWriter, we ensure that no files are actually written during this phase
-            Project project = KubernetesCommonHelper.createProject(applicationInfo, artifactPath);
-            final SessionWriter sessionWriter = new SimpleFileWriter(project, false);
-            final SessionReader sessionReader = new SimpleFileReader(
-                    project.getRoot().resolve("src").resolve("main").resolve("kubernetes"), kubernetesDeploymentTargets
-                            .getEntriesSortedByPriority().stream()
-                            .map(DeploymentTargetEntry::getName).collect(Collectors.toSet()));
-            sessionWriter.setProject(project);
+            Optional<Project> optionalProject = KubernetesCommonHelper.createProject(applicationInfo, artifactPath);
+            optionalProject.ifPresent(project -> {
 
-            if (launchMode.getLaunchMode() != LaunchMode.NORMAL) {
-                // needed for a fresh run
-                Session.clearSession();
-            }
+                final Map<String, String> generatedResourcesMap;
+                final SessionWriter sessionWriter = new SimpleFileWriter(project, false);
+                final SessionReader sessionReader = new SimpleFileReader(
+                        project.getRoot().resolve("src").resolve("main").resolve("kubernetes"), kubernetesDeploymentTargets
+                                .getEntriesSortedByPriority().stream()
+                                .map(DeploymentTargetEntry::getName).collect(Collectors.toSet()));
+                sessionWriter.setProject(project);
 
-            final Session session = Session.getSession(new NoopLogger());
-
-            session.setWriter(sessionWriter);
-            session.setReader(sessionReader);
-
-            session.feed(Maps.fromProperties(config));
-
-            //We need to verify to filter out anything that doesn't extend the Configurator class.
-            //The ConfiguratorBuildItem is a wrapper to Object.
-            allConfigurators.stream().filter(d -> d.matches(Configurator.class)).forEach(i -> {
-                Configurator configurator = (Configurator) i.getConfigurator();
-                session.configurators().add(configurator);
-            });
-
-            //We need to verify to filter out anything that doesn't extend the Decorator class.
-            //The DecoratorBuildItem is a wrapper to Object.
-            allDecorators.stream().filter(d -> d.matches(Decorator.class)).forEach(i -> {
-                String group = i.getGroup();
-                Decorator decorator = (Decorator) i.getDecorator();
-                if (Strings.isNullOrEmpty(group)) {
-                    session.resources().decorate(decorator);
-                } else {
-                    session.resources().decorate(group, decorator);
+                if (launchMode.getLaunchMode() != LaunchMode.NORMAL) {
+                    // needed for a fresh run
+                    Session.clearSession();
                 }
-            });
 
-            // write the generated resources to the filesystem
-            generatedResourcesMap = session.close();
-            List<String> generatedFileNames = new ArrayList<>(generatedResourcesMap.size());
-            for (Map.Entry<String, String> resourceEntry : generatedResourcesMap.entrySet()) {
-                Path path = Paths.get(resourceEntry.getKey());
-                //We need to ignore the config yml
-                if (!path.toFile().getParentFile().getName().equals("dekorate")) {
-                    continue;
-                }
-                String fileName = path.toFile().getName();
-                Path targetPath = outputTarget.getOutputDirectory().resolve(KUBERNETES).resolve(fileName);
-                String relativePath = targetPath.toAbsolutePath().toString().replace(root.toAbsolutePath().toString(), "");
+                final Session session = Session.getSession(new NoopLogger());
 
-                resourceEntry.getKey().replace(root.toAbsolutePath().toString(), KUBERNETES);
-                if (fileName.endsWith(".yml") || fileName.endsWith(".json")) {
-                    String target = fileName.substring(0, fileName.lastIndexOf("."));
-                    if (!deploymentTargets.contains(target)) {
+                session.setWriter(sessionWriter);
+                session.setReader(sessionReader);
+
+                session.feed(Maps.fromProperties(config));
+
+                //We need to verify to filter out anything that doesn't extend the Configurator class.
+                //The ConfiguratorBuildItem is a wrapper to Object.
+                allConfigurators.stream().filter(d -> d.matches(Configurator.class)).forEach(i -> {
+                    Configurator configurator = (Configurator) i.getConfigurator();
+                    session.configurators().add(configurator);
+                });
+
+                //We need to verify to filter out anything that doesn't extend the Decorator class.
+                //The DecoratorBuildItem is a wrapper to Object.
+                allDecorators.stream().filter(d -> d.matches(Decorator.class)).forEach(i -> {
+                    String group = i.getGroup();
+                    Decorator decorator = (Decorator) i.getDecorator();
+                    if (Strings.isNullOrEmpty(group)) {
+                        session.resources().decorate(decorator);
+                    } else {
+                        session.resources().decorate(group, decorator);
+                    }
+                });
+
+                // write the generated resources to the filesystem
+                generatedResourcesMap = session.close();
+                List<String> generatedFileNames = new ArrayList<>(generatedResourcesMap.size());
+                for (Map.Entry<String, String> resourceEntry : generatedResourcesMap.entrySet()) {
+                    Path path = Paths.get(resourceEntry.getKey());
+                    //We need to ignore the config yml
+                    if (!path.toFile().getParentFile().getName().equals("dekorate")) {
                         continue;
                     }
+                    String fileName = path.toFile().getName();
+                    Path targetPath = outputTarget.getOutputDirectory().resolve(KUBERNETES).resolve(fileName);
+                    String relativePath = targetPath.toAbsolutePath().toString().replace(root.toAbsolutePath().toString(), "");
+
+                    resourceEntry.getKey().replace(root.toAbsolutePath().toString(), KUBERNETES);
+                    if (fileName.endsWith(".yml") || fileName.endsWith(".json")) {
+                        String target = fileName.substring(0, fileName.lastIndexOf("."));
+                        if (!deploymentTargets.contains(target)) {
+                            continue;
+                        }
+                    }
+
+                    generatedFileNames.add(fileName);
+                    generatedResourceProducer.produce(
+                            new GeneratedFileSystemResourceBuildItem(
+                                    // we need to make sure we are only passing the relative path to the build item
+                                    relativePath,
+                                    resourceEntry.getValue().getBytes(StandardCharsets.UTF_8)));
                 }
 
-                generatedFileNames.add(fileName);
-                generatedResourceProducer.produce(
-                        new GeneratedFileSystemResourceBuildItem(
-                                // we need to make sure we are only passing the relative path to the build item
-                                relativePath,
-                                resourceEntry.getValue().getBytes(StandardCharsets.UTF_8)));
-            }
-
-            if (!generatedFileNames.isEmpty()) {
-                log.debugf("Generated the Kubernetes manifests: '%s' in '%s'", String.join(",", generatedFileNames),
-                        outputTarget.getOutputDirectory() + File.separator + KUBERNETES);
-            }
-
-            try {
-                if (root != null && root.toFile().exists()) {
-                    FileUtil.deleteDirectory(root);
+                if (!generatedFileNames.isEmpty()) {
+                    log.debugf("Generated the Kubernetes manifests: '%s' in '%s'", String.join(",", generatedFileNames),
+                            outputTarget.getOutputDirectory() + File.separator + KUBERNETES);
                 }
-            } catch (IOException e) {
-                log.debug("Unable to delete temporary directory " + root, e);
+
+                try {
+                    if (root != null && root.toFile().exists()) {
+                        FileUtil.deleteDirectory(root);
+                    }
+                } catch (IOException e) {
+                    log.debug("Unable to delete temporary directory " + root, e);
+                }
+
+            });
+
+            if (!optionalProject.isPresent()) {
+                log.warn("No project was detected, skipping generation of kubernetes manifests!");
             }
         } catch (Exception e) {
             if (launchMode.getLaunchMode() == LaunchMode.NORMAL) {
@@ -205,5 +214,6 @@ class KubernetesProcessor {
 
             log.warn("Failed to generate Kubernetes resources", e);
         }
+
     }
 }

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftProcessor.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftProcessor.java
@@ -125,7 +125,7 @@ public class OpenshiftProcessor {
         List<DecoratorBuildItem> result = new ArrayList<>();
         String name = ResourceNameUtil.getResourceName(config, applicationInfo);
 
-        Project project = KubernetesCommonHelper.createProject(applicationInfo, outputTarget, packageConfig);
+        Optional<Project> project = KubernetesCommonHelper.createProject(applicationInfo, outputTarget, packageConfig);
         result.addAll(KubernetesCommonHelper.createDecorators(project, OPENSHIFT, name, config, metricsConfiguration,
                 annotations, labels, command,
                 ports, livenessPath, readinessPath, roles, roleBindings));

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VanillaKubernetesProcessor.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VanillaKubernetesProcessor.java
@@ -99,7 +99,7 @@ public class VanillaKubernetesProcessor {
         final List<DecoratorBuildItem> result = new ArrayList<>();
         final String name = ResourceNameUtil.getResourceName(config, applicationInfo);
 
-        Project project = KubernetesCommonHelper.createProject(applicationInfo, outputTarget, packageConfig);
+        Optional<Project> project = KubernetesCommonHelper.createProject(applicationInfo, outputTarget, packageConfig);
         result.addAll(KubernetesCommonHelper.createDecorators(project, KUBERNETES, name, config, metricsConfiguration,
                 annotations, labels, command, ports, livenessPath, readinessPath, roles, roleBindings));
 

